src/models/thread.ts

```diff
@@ -193,4 +198,25 @@ export class Thread extends EventEmitter {
     public has(eventId: string): boolean {
         return this.timelineSet.findEventById(eventId) instanceof MatrixEvent;
     }
+
+    public on(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.on(event, listener);
+        return this;
+    }
+    public once(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.once(event, listener);
+        return this;
+    }
+    public off(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.off(event, listener);
+        return this;
+    }
+    public addListener(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.addListener(event, listener);
+        return this;
+    }
+    public removeListener(event: ThreadEvents, listener: (...args: any[]) => void): this {
+        super.removeListener(event, listener);
+        return this;
+    }
```

Comment:
Wonder if we can do this with a generic base class or something, as this is kinda boilerplatey, e.g `EventEmitter<ThreadEvents>` etc

Reply:
I think we can definitely do this. Is it useful to do it within this PR?

Reply:
Up to you, I already approved